src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php

```diff
@@ -60,7 +60,20 @@ public function execute()
 
             $con = $this->connection->getResource();
 
-            $this->search = @ldap_search(
+            switch ($this->options['scope']) {
+                case static::SCOPE_BASE:
+                    $func = 'ldap_read';
+                    break;
+                case static::SCOPE_ONELEVEL:
+                    $func = 'ldap_list';
+                    break;
+                case static::SCOPE_SUBTREE:
+                default:
```

Comment:
Okay. So throw an exception in the default, and set $func to lday_search in the SCOPE_SUBTREE case. :)